src/compiler/parser.ts

```diff
@@ -6055,8 +6057,7 @@ namespace ts {
                 node.name = identifierName;
             }
             if (kind === SyntaxKind.ImportSpecifier && checkIdentifierIsKeyword) {
-                // Report error identifier expected
-                parseErrorAtPosition(checkIdentifierStart, checkIdentifierEnd - checkIdentifierStart, Diagnostics.Identifier_expected);
+                parseErrorAtRange({ pos: checkIdentifierStart, end: checkIdentifierEnd }, Diagnostics.Identifier_expected);

```

Comment:
same here, this basically gives you named arguments in exchange for having to create a temp object.